repositories/blob_repository.go

```diff
@@ -65,10 +65,10 @@ func (repository *blobRepository) openBlobBucket(ctx context.Context, bucketUrl
 	)
 	defer span.End()
 
-	if repository.buckets[bucketUrl] == nil {
-		repository.m.Lock()
-		defer repository.m.Unlock()
+	repository.m.Lock()
```

Comment:
A quick note on this change.

I noticed lock contention when multiple goroutine were trying to open a bucket that had not been initialized before. In our case, 200 goroutine tried to acquire the lock **after** having checked that it was `nil`, repeating the costly initialization setup down there.

For offloading, it meant that the first batch would take up to 15 seconds instead of 1.

I am not entirely happy with this, since now we get contention of every access, even though we safeguard against initializing everything again (all except the first one should be short enough, as opposed to the previous iteration, where all concurrent 200 would be slow and serialized).